gpm_api/tests/test_utils/test_geospatial.py

```diff
@@ -42,3 +40,378 @@
+        geospatial.get_country_extent(country)
+
+
+def test_get_continent_extent(
+    continent_extent_dictionary: ExtentDictionary,
+) -> None:
+    """Test get_continent_extent"""
+
+    # Test valid continent
+    continent = "Africa"
+    expected_extent = continent_extent_dictionary[continent]
+    returned_extent = geospatial.get_continent_extent(continent)
+    assert returned_extent == expected_extent
+
+    # Test invalid continent
+    continent = "Invalid"
+    with pytest.raises(ValueError):
+        geospatial.get_continent_extent(continent)
+
+    # Test typo in continent name
+    continent = "Arica"
+    with pytest.raises(ValueError) as exception_info:
+        geospatial.get_continent_extent(continent)
+
+    assert "Africa" in str(exception_info.value.args[0])
+
+    # Test invalid continent type
+    continent = 123
+    with pytest.raises(TypeError):
+        geospatial.get_continent_extent(continent)
+
+
+def test_get_extent() -> None:
+    """Test get_extent"""
+
+    ds = xr.Dataset(
+        {
+            "lon": [-10, 0, 20],
+            "lat": [-30, 0, 40],
+        }
+    )
+
+    # Test without padding
+    expected_extent = (-10, 20, -30, 40)
+    returned_extent = geospatial.get_extent(ds)
+    assert returned_extent == expected_extent
+
+    # Test with float padding
+    padding = 0.1
+    expected_extent = (-10.1, 20.1, -30.1, 40.1)
+    returned_extent = geospatial.get_extent(ds, padding=padding)
+    assert returned_extent == expected_extent
+
+    # Test with padding exceeding bounds
+    padding = 180
+    expected_extent = (-180, 180, -90, 90)
+    returned_extent = geospatial.get_extent(ds, padding=padding)
+    assert returned_extent == expected_extent
+
+    # Test with tuple padding
+    padding = (0.1, 0.2)
+    expected_extent = (-10.1, 20.1, -30.2, 40.2)
+    returned_extent = geospatial.get_extent(ds, padding=padding)
+    assert returned_extent == expected_extent
+
+    # Test with invalid padding
+    with pytest.raises(TypeError):
+        geospatial.get_extent(ds, padding="invalid")
+    with pytest.raises(ValueError):
+        geospatial.get_extent(ds, padding=(0.1,))
+    with pytest.raises(ValueError):
+        geospatial.get_extent(ds, padding=(0.1, 0.2, 0.3))
+
+    # Test with object crossing dateline (not implemented)
+    # ds = xr.Dataset({
+    #     "lon": [170, 180, -160],
+    #     "lat": [-30, 0, 40],
+    # })
+    # with pytest.raises(NotImplementedError):
+    #     geospatial.get_extent(ds)
+
+
+@pytest.fixture
+def orbit_dataset() -> xr.Dataset:
+    # Values along track
+    lon = np.arange(-50, 51, 10)
+    lat = np.zeros_like(lon)
+
+    # Add cross track dimension
+    lat = lat[np.newaxis, :]
+    lon = lon[np.newaxis, :]
+
+    # Create dataset
+    ds = xr.Dataset()
+    ds["lat"] = (("cross_track", "along_track"), lat)
+    ds["lon"] = (("cross_track", "along_track"), lon)
+
+    return ds
+
+
+@pytest.fixture
+def orbit_dataset_multiple_crossings() -> xr.Dataset:
```

Comment:
`antimeridian_crossing`?

Reply:
Multiple passes over of the extent. Should I change the name?

Reply:
I see that you repeat the array twice. I guess you want to test the dateline crossing.
Maybe you could make the test data more realistic. 
Have latitude ascending and descending (between -90 and 90) and longitude going between -180 and 180 (and repeating). 

You can change the name, i.e. `multiple_antimeridian_crossing_orbit`.

I guess you will end up creating other such sort of orbit fixtures when testing the orbit plotting routines. 
Fine for me to postpone stuffs if you prefer ;)